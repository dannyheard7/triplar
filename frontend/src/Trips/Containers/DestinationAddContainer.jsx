import React from "react";
import DestinationForm from "Trips/Components/DestinationForm";
import api from "utils/api.js"
import FormContainer from "Forms/Containers/FormContainer";

export default class DestinationAddContainer extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            showDestinationAdd: false
        };

        this.onClick = this.onClick.bind(this);
        this.onSuccess = this.onSuccess.bind(this)
    }

    onClick() {
        this.setState({showDestinationAdd: !this.state.showDestinationAdd});
    }

    onSuccess(itinerary) {
        this.setState({showDestinationAdd: false});
        this.props.onSuccess(itinerary)
    }

    apiFunc = api.curryAPIFunc(api.addDestinationToTrip, this.props.tripId);

    render () {
        return (
            <div>
                <div className="row">
                    <div className="col md-2">
                        <button className="btn btn-primary" onClick={this.onClick}>Add Destination</button>
                    </div>
                </div>
                {this.state.showDestinationAdd &&
                    <div className="card">
                        <FormContainer onSuccess={this.onSuccess} apiFunction={this.apiFunc}>
                            <DestinationForm submitLabel="Add"/>
                        </FormContainer>
                    </div>
                }
            </div>
        )
    }
}