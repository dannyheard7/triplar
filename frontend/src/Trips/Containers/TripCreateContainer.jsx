import React from "react";

import TripForm from "Trips/Components/TripForm";
import ReduxFormContainer from "../../Forms/Containers/ReduxFormContainer";
import {connect} from "react-redux";
import {createTrip} from "../utils/actions";

export class TripCreateContainer extends React.Component {
    constructor(props) {
        super(props);

        this.onSuccess = this.onSuccess.bind(this);
        this.onClick = this.onClick.bind(this);

        this.state = {
            showTripCreate: false
        };
    }

    onSuccess(data) {
        this.setState({showTripCreate: false});

    };

    onClick() {
        this.setState({showTripCreate: !this.state.showTripCreate});
    }

    render () {
        return (
            <div>
                <button className="btn btn-primary" onClick={this.onClick}>New Trip</button>
                {this.state.showTripCreate &&
                    <div className="card">
                        <ReduxFormContainer errors={this.props.errors} action={createTrip}>
                            <TripForm submitLabel="Create" />
                        </ReduxFormContainer>
                    </div>
                }
            </div>
        )
    }
}

const mapStateToProps = (state) => {return {errors: state.trips.errors}};
export default connect(mapStateToProps)(TripCreateContainer)