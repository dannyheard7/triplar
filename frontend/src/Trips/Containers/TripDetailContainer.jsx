import React from "react";

import TripDetail from "Trips/Components/TripDetail";
import {Link} from "react-router-dom";
import LoadingIndicator from "App/Components/LoadingIndicator";
import {connect} from "react-redux";
import LocationListContainer from "../../Itinerary/Containers/TripLocationListContainer";
import LocationAddContainer from "../../Itinerary/Containers/LocationAddContainer";

export class TripDetailContainer extends React.Component {
    render() {
        const trip = this.props.trip;

        if (!trip) {
            return <LoadingIndicator />;
        } else {
            if(this.props.locations) {
                trip.locations = this.props.locations;
            }

            return (
                <div>
                    <TripDetail trip={trip}/>
                    <Link to={`/trips/${trip.id}/edit`}><button className="btn btn-light">Edit</button></Link>
                    <Link to={`/trips/${trip.id}/delete`}><button className="btn btn-danger">Delete Trip</button></Link>
                    <LocationListContainer tripId={trip.id}/>
                    <LocationAddContainer tripId={trip.id} />
                </div>
            );
        }
    }
}

function mapStateToProps(state, ownProps) {
  return {
      trip: state.trips.trips.find(x => x.id === ownProps.match.params.id),
      locations: state.itineraries.locationItineraries.filter(x => x.trip.id === ownProps.match.params.id).map(x => { return {city: x.city} })
  };
}

export default connect(mapStateToProps)(TripDetailContainer)