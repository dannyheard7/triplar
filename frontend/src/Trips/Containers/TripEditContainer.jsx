import React from "react";

import TripForm from "Trips/Components/TripForm.jsx";
import { connect } from "react-redux";
import {editTrip} from "../utils/actions";
import ReduxFormContainer from "../../Forms/Containers/ReduxFormContainer";

export class TripEditContainer extends React.Component {
    curryAction = (action, id) => (object) => action(id, object);

    render() {
        let action = this.curryAction(editTrip, this.props.trip.id);

        return (
            <div className="card">
                <ReduxFormContainer action={action} >
                    <TripForm submitLabel="Save" trip={this.props.trip}/>
                </ReduxFormContainer>
            </div>
        );
    }
}

TripEditContainer.defaultProps = {
    trip: {}
};

function mapStateToProps(state, ownProps) {
  return {
      trip: state.trips.trips.find(x => x.id === ownProps.match.params.id)
  };
}

export default connect(mapStateToProps)(TripEditContainer)