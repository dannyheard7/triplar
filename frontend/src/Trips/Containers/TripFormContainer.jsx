import React from "react";
import TripForm from "Trips/Components/TripForm";

export default class TripFormContainer extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            trip: this.props.trip,
            errors: []
        };

        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }


    handleChange(event) {
        let trip = this.state.trip;
        trip[event.target.name] = event.target.value;

        this.setState({trip: trip});
    }

    handleSubmit(e) {
        e.preventDefault();

        this.props.apiFunction(this.state.trip).then(response => {
            if(response.status === 400) {
                this.setState({errors: response.data})
            } else {
                this.setState({trip: {}, errors: []});
                this.props.onSuccess(response.data);
            }
        });
    }

    render () {
        return (
            <TripForm onChange={this.handleChange} onSubmit={this.handleSubmit} submitLabel={this.props.submitLabel}
                      errors={this.state.errors} suggestions={this.state.suggestions} trip={this.state.trip} />
        )
    }

}

TripFormContainer.defaultProps = {
    trip: {}
};