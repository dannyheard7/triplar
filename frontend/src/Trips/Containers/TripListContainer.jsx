import React from "react";

import TripList from "Trips/Components/TripList";
import {connect} from "react-redux";
import {getTrips} from "../utils/actions";

export class TripListContainer extends React.Component {
    componentDidMount() {
        this.props.dispatch(getTrips())
    }

    render() {
        return (<TripList trips={this.props.trips}/>);
    }
}

const mapStateToProps = state => ({
  trips: state.trips.trips,
});

export default connect(mapStateToProps)(TripListContainer)