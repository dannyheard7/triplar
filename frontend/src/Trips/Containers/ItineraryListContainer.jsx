import React from "react";
import DestinationItinerary from "../Components/DestinationItinerary";
import DestinationAddContainer from "Trips/Containers/DestinationAddContainer";

export default class ItineraryListContainer extends React.Component {
     constructor(props) {
        super(props);

        this.state = {
            itineraries: this.props.itineraries
        };

        this.onDestinationAdd = this.onDestinationAdd.bind(this);
    }

    onDestinationAdd(destination) {
         this.setState({itineraries: this.state.itineraries.concat([destination])})
    }

    render() {
        const itineraries = this.state.itineraries.map((itinerary) =>
            <DestinationItinerary key={itinerary.id} itinerary={itinerary}/>
        );

        return (
            <div>
                {itineraries}
                <DestinationAddContainer tripId={this.props.tripId} onSuccess={this.onDestinationAdd}/>
            </div>
        );
    }
}