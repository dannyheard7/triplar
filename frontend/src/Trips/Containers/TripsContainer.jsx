import React from "react";

import TripListContainer from "Trips/Containers/TripListContainer";
import TripCreateContainer from "Trips/Containers/TripCreateContainer";

// TODO: Move into a functional component

export default class TripsContainer extends React.Component {
     render() {
        return (
            <div>
                <TripListContainer />
                <TripCreateContainer />
            </div>
        )
    }
}