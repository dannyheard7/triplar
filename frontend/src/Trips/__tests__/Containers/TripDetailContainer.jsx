import React from "react";
import {shallow} from "enzyme";
import {TripDetailContainer} from "Trips/Containers/TripDetailContainer";
import LoadingIndicator from "App/Components/LoadingIndicator";

import ShallowRenderer from 'react-test-renderer/shallow';
const renderer = new ShallowRenderer();
const faker = require('faker');

const api = require('Trips/utils/trips.api.js');

describe('<TripDetailContainer />', () => {
    const props = {
        trip: {'id': faker.random.number(), 'name': faker.random.word(), 'created_by': faker.internet.email()},
        match: {params: {id: faker.random.number()}}
    };

    test('renders loading text when no trip is passed', async () => {
        let container = shallow(<TripDetailContainer {...props} trip={null} />);
        expect(container.find(LoadingIndicator).length).toEqual(1);
        container = shallow(<TripDetailContainer {...props}  />);
        expect(container.find(LoadingIndicator).length).toEqual(0);
    });


    // Move to redux tests
    test.skip('response from api call is set as state', async () => {
        const stub = jest.spyOn(api.default, "getTripDetail");
        const trip = {'id': faker.random.number(), 'name': faker.random.word(), 'created_by': faker.internet.email()};
        stub.mockReturnValueOnce(Promise.resolve({status: 200, data: {data: {trip: trip}}}));

        const container = shallow(<TripDetailContainer {...props} match={{params: {id: trip.id}}} />);
        await Promise.resolve();

        expect(stub).toBeCalledWith(trip.id);
        expect(container.state('trip')).toEqual(trip);

        stub.mockRestore(); // Changed the mock so restore it to original
    });

    test.skip("Renders trip not found on trips that don't exist", () => {

    });

    test('With a trip object renders correctly', () => {
        const trip = {'id': 1, 'name': "New Trip", 'created_by': "Danny Heard"};
        const result = renderer.render(<TripDetailContainer trip={trip} />);
        expect(result).toMatchSnapshot();
    });

});