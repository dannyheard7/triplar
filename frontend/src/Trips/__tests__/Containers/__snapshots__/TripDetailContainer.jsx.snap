// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TripDetailContainer /> With a trip object renders correctly 1`] = `
<div>
  <TripDetail
    trip={
      Object {
        "created_by": "Danny Heard",
        "id": 1,
        "name": "New Trip",
      }
    }
  />
  <Link
    replace={false}
    to="/trips/1/edit"
  >
    <button
      className="btn btn-light"
    >
      Edit
    </button>
  </Link>
  <Link
    replace={false}
    to="/trips/1/delete"
  >
    <button
      className="btn btn-danger"
    >
      Delete Trip
    </button>
  </Link>
  <Connect(TripLocationListContainer)
    tripId={1}
  />
  <LocationAddContainer
    tripId={1}
  />
</div>
`;
